core/tools.js

```diff
@@ -1518,6 +1518,87 @@
 		},
 
 		/**
+		 * Convert hex string to array containing 1 byte in each cell. Bytes are represent as an Integer numbers.
+		 *
+		 * @since 4.8.0
+		 * @param {String} hexString Contains input string which represent bytes, e.g. `"08A11D8ADA2B"`.
+		 * @returns {Array} Bytes stored in a form of Integer numbers, e.g. `[ 8, 161, 29, 138, 218, 43 ]`.
+		 */
+		convertHexStringToBytes: function( hexString ) {
+			var bytesArray = [],
+				bytesArrayLength = hexString.length / 2,
+				i;
+
+			for ( i = 0; i < bytesArrayLength; i++ ) {
+				bytesArray.push( parseInt( hexString.substr( i * 2, 2 ), 16 ) );
+			}
+			return bytesArray;
+		},
+
+		/**
+		 * Convert bytes array into base64 code.
+		 *
+		 * Bytes are `8bit` numbers, where base64 use `6bit` to store data. That's why we process 3 Bytes into 4 characters representing base64.
+		 *
+		 * Algorithm:
+		 *
+		 * 1. Take `3 * 8bit`.
+		 * 2. If there is less than 3 byte, fill empty bits with zeros.
+		 * 3. Transform `3 * 8bit` into `4 * 6bit` numbers.
+		 * 4. Translate those numbers to proper character related to base64.
+		 * 5. If extra zeros bytes were added fill them with `=` sign.
+		 *
+		 * Example:
+		 *
+		 * * Bytes Array: [ 8, 161, 29, 138, 218, 43 ] -> binary: `0000 1000 1010 0001 0001 1101 1000 1010 1101 1010 0010 1011`.
+		 * * Binary: `0000 10|00 1010| 0001 00|01 1101| 1000 10|10 1101| 1010 00|10 1011` ← `|` (pipe) shows where base64 will cut bits during transformation.
+		 * * Now we have 6bit numbers (written in decimal values), which are translated to indexes in `base64characters` array.<br />
+		 * Decimal: `2 10 4 29 34 45 40 43` → base64: `CKEditor`.
+		 *
+		 * @since 4.8.0
+		 * @param bytesArray Array which stores 1 byte in each cell as Integer number.
+		 * @returns Base64 string which represents input bytes.
```

Comment:
There is no such type as `bytesArray` or `Base64`, it should probably be `Array` and `String`. You can include more information (that it is array of bytes / string in base64 format) in the `@param`/`@returns` description. Also remember about brackets when defining type - `{Array}`.